sonar-batch/src/main/java/org/sonar/batch/report/CoveragePublisher.java

```diff
@@ -0,0 +1,130 @@
+        @Override
+        public void apply(String value, Coverage.Builder builder) {
+          builder.setUtHits(Integer.parseInt(value) > 0);
+        }
+      });
+      applyLineMeasure(resource.key(), ((InputFile) resource.inputPath()).lines(), CoreMetrics.CONDITIONS_BY_LINE_KEY, coveragePerLine, new MeasureOperation() {
+        @Override
+        public void apply(String value, Coverage.Builder builder) {
+          builder.setConditions(Integer.parseInt(value));
+        }
+      });
+      applyLineMeasure(resource.key(), ((InputFile) resource.inputPath()).lines(), CoreMetrics.COVERED_CONDITIONS_BY_LINE_KEY, coveragePerLine, new MeasureOperation() {
+        @Override
+        public void apply(String value, Coverage.Builder builder) {
+          builder.setUtCoveredConditions(Integer.parseInt(value));
+        }
+      });
+      applyLineMeasure(resource.key(), ((InputFile) resource.inputPath()).lines(), CoreMetrics.IT_COVERAGE_LINE_HITS_DATA_KEY, coveragePerLine, new MeasureOperation() {
+        @Override
+        public void apply(String value, Coverage.Builder builder) {
+          builder.setItHits(Integer.parseInt(value) > 0);
+        }
+      });
+      applyLineMeasure(resource.key(), ((InputFile) resource.inputPath()).lines(), CoreMetrics.IT_COVERED_CONDITIONS_BY_LINE_KEY, coveragePerLine, new MeasureOperation() {
+        @Override
+        public void apply(String value, Coverage.Builder builder) {
+          builder.setItCoveredConditions(Integer.parseInt(value));
+        }
+      });
+      applyLineMeasure(resource.key(), ((InputFile) resource.inputPath()).lines(), CoreMetrics.OVERALL_COVERED_CONDITIONS_BY_LINE_KEY, coveragePerLine, new MeasureOperation() {
+        @Override
+        public void apply(String value, Coverage.Builder builder) {
+          builder.setOverallCoveredConditions(Integer.parseInt(value));
+        }
+      });
+
+      writer.writeFileCoverage(resource.batchId(), Iterables.transform(coveragePerLine.values(), new Function<Coverage.Builder, Coverage>() {
+        @Override
+        public Coverage apply(Builder input) {
+          return input.build();
+        }
+      }));
+    }
+  }
+
+  void applyLineMeasure(String inputFileKey, int lineCount, String metricKey, Map<Integer, Coverage.Builder> coveragePerLine, MeasureOperation op) {
+    Iterable<Measure> measures = measureCache.byMetric(inputFileKey, metricKey);
+    if (measures.iterator().hasNext()) {
+      Measure measure = measures.iterator().next();
+      Map<Integer, String> lineMeasures = KeyValueFormat.parseIntString((String) measure.value());
+      for (Map.Entry<Integer, String> lineMeasure : lineMeasures.entrySet()) {
+        int lineIdx = lineMeasure.getKey();
+        if (lineIdx <= lineCount) {
+          String value = lineMeasure.getValue();
+          if (StringUtils.isNotEmpty(value)) {
```

Comment:
:new: ![MAJOR](https://raw.githubusercontent.com/henryju/image-hosting/master/major.png)Refactor this code to not nest more than 3 if/for/while/switch/try statements.[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)